File: src/main/java/com/gmail/hexragon/gn4rBot/command/fun/ASCIICommand.java
package com.gmail.hexragon.gn4rBot.command.fun;

import com.gmail.hexragon.gn4rBot.managers.commands.Command;
import com.gmail.hexragon.gn4rBot.managers.commands.CommandExecutor;
import net.dv8tion.jda.entities.Message;
import org.apache.commons.lang3.StringUtils;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.nodes.Node;
import org.jsoup.nodes.TextNode;

import java.util.List;

@Command(
		aliases = {"ascii"},
		usage = "(string)",
		description = "ASCII text art!"
)
public class ASCIICommand extends CommandExecutor
{
	@Override
	public void execute(Message message, String[] args)
	{
		if (args.length == 0)
		{
			message.getChannel().sendMessage(String.format("%s ➜ Please provide a query.", message.getAuthor().getAsMention()));
			return;
		}
		
		try
		{
			String query = StringUtils.join(args, "%20");
			
			Document document;
			
			document = Jsoup.connect("http://ascii-text.com/online-ascii-banner-text-generator/slant/"+query).get();
			
			Element element = document.getElementById("cb").getElementsByTag("pre").get(0);
			
			String builder = "```\n" + getText(element) + "```";
			
			message.getChannel().sendMessage(builder);
			
		}
		catch (Exception e)
		{
			message.getChannel().sendMessage(String.format("%s ➜ Unable to generate ASCII banner.", message.getAuthor().getAsMention()));
			e.printStackTrace();
		}
	}
	
	private String getText(Element cell) {
		String text = null;
		List<Node> childNodes = cell.childNodes();
		if (childNodes.size() > 0) {
			Node childNode = childNodes.get(0);
			if (childNode instanceof TextNode) {
				text = ((TextNode)childNode).getWholeText();
			}
		}
		if (text == null) {
			text = cell.text();
		}
		return text;
	}
}

File: src/main/java/com/gmail/hexragon/gn4rBot/command/fun/ASCIICommand.kt
package com.gmail.hexragon.gn4rBot.command.`fun`

import com.gmail.hexragon.gn4rBot.managers.commands.CommandExecutor
import com.gmail.hexragon.gn4rBot.managers.commands.annotations.Command
import com.gmail.hexragon.gn4rBot.util.GnarMessage
import org.apache.commons.lang3.StringUtils
import org.jsoup.Jsoup
import org.jsoup.nodes.Element
import org.jsoup.nodes.TextNode

@Command(aliases = arrayOf("ascii"), usage = "(string)", description = "ASCII text art!", showInHelp = false)
class ASCIICommand : CommandExecutor()
{
    override fun execute(message : GnarMessage, args : Array<String>)
    {
        if (args.size == 0)
        {
            message.reply("Please provide a query.")
            return
        }
        
        try
        {
            val query = StringUtils.join(args, "+")
            
            if (query.length > 15)
            {
                message.reply("The query has too many characters. `15 at most.`")
                return
            }
            
            val document = Jsoup.connect("http://artii.herokuapp.com/make?text=$query").get()
            
            println(document.toString())
            
            val element = document.getElementsByTag("body")[0]
            
            val builder = "```\n${getText(element)}```"
            
            message.replyRaw(builder)
        }
        catch (e : Exception)
        {
            message.reply("Unable to generate ASCII art. `Developer Note: Our ASCII API is down for the moment.`")
            e.printStackTrace()
        }
        
    }
    
    private fun getText(cell : Element) : String?
    {
        var text : String? = null
        val childNodes = cell.childNodes()
        if (childNodes.size > 0)
        {
            val childNode = childNodes[0]
            if (childNode is TextNode)
            {
                text = childNode.wholeText
            }
        }
        if (text == null)
        {
            text = cell.text()
        }
        return text
    }
}

